kiva/agg/tests/test_gcmem.py

```diff
@@ -6,8 +6,14 @@
 import gc as garbagecollector
 
 from kiva.image import GraphicsContext, GraphicsContextSystem
-from etsdevtools.debug.memusage import get_mem_usage
+try:
+    from etsdevtools.debug.memusage import get_mem_usage
+    etsdevtools_available = True
+except ModuleNotFoundError:
+    etsdevtools_available = False
 
+
+@unittest.skipIf(not etsdevtools_available, "test requires etsdevtools")
```

Comment:
I couldn't find the skip message in the log. I believe that's because test discovery does not use the `test_suite` and `test` functions below, and none of the methods in this test case has a name with the prefix `test_`: they start with the prefix `check_` instead.  Not sure if we still need to keep this `check_` prefix.  Could we remove the `test_suite`, `test`  and the `__main__` block below, and rename the method to `test_...` please?  Such change would be within the scope of this PR because this PR changes how tests are discovered. The tests will still be skipped, though.

Reply:
Should I also add `etsdevtools` as a test dependency like what was done for `hypothesis`?

Reply:
Er... `etsdevtools` is a worse dependency here. Let's not go there, and kill this test in a separate PR. This PR should aim at minimal changes in the number of tests being run: the purpose is to switch test runner. It is not meaningful to keep this test given it has not been exercised, but that's orthogonal.

Reply:
opened #483